### ISSUE TITLE
Next 12.0.2 -> Infinite loop on _error page while routing on client side

### ISSUE THREAD

**Issue by Abhishek-Sati** [vercel/next.js#30846]
### What version of Next.js are you using?

12.0.2

### What version of Node.js are you using?

17.0.1

### What browser are you using?

chrome

### What operating system are you using?

macOS Big Sur version 11.2.3(m1 chip)

### How are you deploying your application?

next start

### Describe the Bug

I am using router.replace() function to redirect users from auth page on click of the login button, but the problem is if the page(where I am redirecting the user) has some error inside it, then _error page loads and goes into an infinite loop.

### Expected Behavior

in case if any page has some error then _error should load only once and stop, as it is working in server-side rendering.

### To Reproduce

1. on click of any button, call router.push('/abc').
2. now on page /abc create a bug, write: console.log({}.map(item => item)) (applying map on object will generate an error).
3. now while you click on button, you will be redirected to /abc where you already created a bug which will load _error page, here you will see that _error is reloading with an infinite loop.

**Comment by timneutkens** [comment]
Could you add a repository that reproduces this?

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
